core/base/mergeTreePrincipalGeodesics/MergeTreePrincipalGeodesicsBase.h

```diff
@@ -0,0 +1,893 @@
+/// \ingroup base
+/// \class ttk::MergeTreePrincipalGeodesicsBase
+/// \author Mathieu Pont <mathieu.pont@lip6.fr>
+/// \date 2022
+///
+/// \b Related \b publication: \n
+/// "Principal Geodesic Analysis of Merge Trees (and Persistence Diagrams)" \n
+/// Mathieu Pont, Jules Vidal, Julien Tierny.\n
+
+#pragma once
+
+#include <MergeTreeBase.h>
+#include <MergeTreeDistance.h>
+#include <cmath>
+
+namespace ttk {
+
+  class MergeTreePrincipalGeodesicsBase : virtual public Debug,
+                                          public MergeTreeBase {
+  protected:
+    unsigned int k_ = 10;
+
+    // Filled by the algorithm
+    std::vector<std::vector<std::vector<double>>> vS_, v2s_, trees2Vs_,
+      trees2V2s_;
+    std::vector<std::vector<double>> allTs_, allScaledTs_, allTreesTs_;
+    std::vector<std::vector<double>> branchesCorrelationMatrix_,
+      persCorrelationMatrix_;
+    std::vector<std::vector<std::tuple<ftm::idNode, ftm::idNode, double>>>
+      baryMatchings_, baryMatchings2_;
+
+  public:
+    MergeTreePrincipalGeodesicsBase() {
+      this->setDebugMsgPrefix(
+        "MergeTreePrincipalGeodesicsBase"); // inherited from Debug: prefix will
+                                            // be printed at the beginning of
+                                            // every msg
+    }
+
+    //----------------------------------------------------------------------------
+    // Matching / Distance
+    //----------------------------------------------------------------------------
+    template <class dataType>
+    void computeOneDistance(
+      ftm::MergeTree<dataType> &tree1,
+      ftm::MergeTree<dataType> &tree2,
+      std::vector<std::tuple<ftm::idNode, ftm::idNode, double>> &matching,
+      dataType &distance,
+      bool isCalled = false,
+      bool useDoubleInput = false,
+      bool isFirstInput = true) {
+      MergeTreeDistance mergeTreeDistance;
+      mergeTreeDistance.setDebugLevel(2);
```

Comment:
what if someone wants to run ttk with no debug information at all?
I suggest to use a predefined value (in this example 2), unless `debugLevel_` is already set at a lower value (for instance 1 or 0). Could you address this remark throughout your code?

Reply:
Done in the next commit!